### PR TITLE
Add coverage to file_regex status check method

### LINKED ISSUE #1447: Add coverage for file_regex status check
### Please describe your feature

Add coverage for status check of `file_regex`. See https://app.codecov.io/gh/buildtesters/buildtest/blob/devel/buildtest/buildsystem/checks.py

Screen Shot 2023-04-19 at 10 51 18 PM

### Suggest potential solution

_No response_

### Additional Information

_No response_

### Post question in Slack

- [X] I agree that I posted my question in [slack](https://hpcbuildtest.slack.com/) before creating this issue

### Is there an existing issue

- [X] I confirm there is no existing [issue](https://github.com/buildtesters/buildtest/issues) for this issue

### PR DESCRIPTION
**Pull request by braeeast2001**
Add test_file_regex to test_builders.py using regex_on_filename.yml.

### CODE REVIEW

File: tests/builders/test_builders.py [hunk 1]
```diff
@@ -138,3 +138,15 @@ def test_file_count():
         configuration=config,
     )
     cmd.build()
+
+def test_file_regex():
+    """This test buildspec using status check with 'file_regex'"""
+    cmd = BuildTest(
+        buildspecs=[
+            os.path.join(here, "file_regex.yml")
```
Comment (szuananwar) [inline tests/builders/test_builders.py]: line 146: These is no file_regex.yml file in tutorials/test_status to link to, please make sure the name of yml file is correct.

**Review by szuananwar — COMMENTED** [review]
line 146: These is no file_regex.yml file in tutorials/test_status to link to, please make sure the name of yml file is correct.

File: tests/builders/test_builders.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def test_file_regex():
     """This test buildspec using status check with 'file_regex'"""
     cmd = BuildTest(
         buildspecs=[
-            os.path.join(here, "file_regex.yml")
+            os.path.join(here, "status_regex.yml")
```
Comment (szuananwar) [inline tests/builders/test_builders.py]: Use this test https://github.com/buildtesters/buildtest/blob/devel/tutorials/test_status/regex_on_filename.yml for your regression test.

File: tests/builders/test_builders.py [hunk 3]
```diff
@@ -138,3 +138,15 @@ def test_file_count():
         configuration=config,
     )
     cmd.build()
+
+def test_file_regex():
+    """This test buildspec using status check with 'file_regex'"""
```
Comment (szuananwar) [inline tests/builders/test_builders.py]: Please change the comment here to match with the correct yml file name that will be linked for test.

**Review by szuananwar — COMMENTED** [review]
Please change the comment in 143 to match with the correct yml file name that will be linked for test.

**Review by szuananwar — COMMENTED** [review]
The comment in the methods matches the linked yml file.

File: tests/builders/test_builders.py [hunk 4]
```diff
@@ -138,3 +138,15 @@ def test_file_count():
         configuration=config,
     )
     cmd.build()
+
+def test_file_regex():
+    """This test buildspec using status check with 'regex_on_filename'"""
```
Comment (shahzebsiddiqui) [inline tests/builders/test_builders.py]: So this docstring is not correct. The filename is `regex_on_filename` but that not what the test is doing. The test is performing status check on keyword `file_regex`. See https://buildtest.readthedocs.io/en/devel/buildspecs/buildspec_overview.html#specify-regular-expression-on-arbitrary-filenames

I would rephrase this docstring to 

```python
def test_file_regex():
    """This test will perform status check with regular expression on file via 'file_regex'"""
```

Comment (szuananwar) [reply tests/builders/test_builders.py]: @shahzebsiddiqui, the filename is regex_on_filename and the test is performing on regex_on_filename keyword. She got it and I commented "The comment in the methods matches the linked yml file" 14 hours ago.

Comment (shahzebsiddiqui) [reply tests/builders/test_builders.py]: Yes what you said is correct but the docstring makes no sense it is too technical and tied to the name of file and test name which can change. The actual keyword that is being tested is `file_regex` that is used in buildspec to specify a filename to perform regular expression. I think even removing the key word `file_regex` would be sufficient too

**Comment by szuananwar** [comment]
@braeeast2001 for style check, please run buildtest sytle -a
see this for more info: https://buildtest.readthedocs.io/en/devel/contributing/code_contribution_guide.html

File: tests/builders/test_builders.py (hunk 4 above)
Comment (szuananwar) [reply tests/builders/test_builders.py]: Okay, thanks,
@braeeast2001 please see the above comment by @shahzebsiddiqui and update your docstring.

**Comment by szuananwar** [comment]
@shahzebsiddiqui I think @braeeast2001 first should check if the file is exists in the directory then do a regex check. I think she should use this test https://github.com/buildtesters/buildtest/blob/devel/tutorials/test_status/exists.yml for her regression test. What do you think?

File: tests/builders/test_builders.py [hunk 5]
```diff
@@ -140,11 +140,10 @@ def test_file_count():
     cmd.build()
 
 
-def test_regex_check():
-    """This test buildspec using status check with  'status_regex'"""
+def test_file_regex():
```
Comment (szuananwar) [inline tests/builders/test_builders.py:163]: You deleted Test_regex_check method, you suppose to add your method and not delete the exist one.

File: tutorials/test_status/regex_on_invailds.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  regex_on_invalid_fname:
+    type: script
+    executor: generic.local.bash
+    description: Test regex on an invalid file name
+    run: |
+      echo "Hello" > hello.txt
+    status:
+      file_regex:
+      - file: /fox.txt
+      - exp: '^(Hello)$'
```
Comment (shahzebsiddiqui) [inline tutorials/test_status/regex_on_invailds.yml]: @braeeast2001 good job this looks good now can you add one more test in `regex_on_invalids.yml` that will perform regular expression on valid file but regex is invalid.

Also there is a spelling error in name of file `regex_on_invailds.yml` -> `regex_on_invalids.yml`

File: tests/builders/test_builders.py [hunk 7]
```diff
@@ -151,9 +151,20 @@ def test_multicompilers_with_script_schema():
 
 
 def test_regex_check():
-    """This test buildspec using status check with  'status_regex'"""
+     """This test buildspec using status check with  'status_regex'"""
```
Comment (shahzebsiddiqui) [inline tests/builders/test_builders.py]: change `status_regex` -> `regex` in doc string

File: tutorials/test_status/regex_on_invalids.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    executor: generic.local.bash
+    description: Test regex on an invalid file name
+    run: |
+      echo "Hello" > hello.txt
+    status:
+      file_regex:
+      - file: /fox.txt
+      - exp: '^(Hello)$'
+
+regex_where_validfname_invaildregex:
```
Comment (shahzebsiddiqui) [inline tutorials/test_status/regex_on_invalids.yml]: spelling error here can you change the name `file_regex_with_invalid_expression`

File: tutorials/test_status/regex_on_invalids.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+build specs:
```
Comment (shahzebsiddiqui) [inline tutorials/test_status/regex_on_invalids.yml]: this is invalid buildspec its `build specs` -> `buildspecs`

File: tutorials/test_status/regex_on_invalids.yml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+regex_where_validfname_invaildregex:
+  type: script
+  executor: generic.local.bash
+  description: Test regex on a valid file where regex is invaild
+  run: |
+    echo "Jump" > jump.txt
+  status:
+    file_regex:
+      - file: foo/bar.txt
```
Comment (shahzebsiddiqui) [inline tutorials/test_status/regex_on_invalids.yml]: change this to `jump.txt` we want the file to be valid here

File: tutorials/test_status/regex_on_invalids.yml [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+      file_regex:
+      - file: /fox.txt
+      - exp: '^(Hello)$'
+
+regex_where_validfname_invaildregex:
+  type: script
+  executor: generic.local.bash
+  description: Test regex on a valid file where regex is invaild
+  run: |
+    echo "Jump" > jump.txt
```
Comment (shahzebsiddiqui) [inline tutorials/test_status/regex_on_invalids.yml]: please put this on a single line `run: echo "Jump" > jump.txt`

If its multi line then do what you did

**Review by szuananwar — COMMENTED** [review]
Thanks @braeeast2001 for making all the necessary changes. 
The style check is failed, please run buildtest sytle -a
and see this for more info: https://buildtest.readthedocs.io/en/devel/contributing/code_contribution_guide.html

File: tests/builders/test_builders.py [hunk 12]
```diff
@@ -151,21 +151,22 @@ def test_multicompilers_with_script_schema():
...
 
 
 def test_file_regex():
     """This test will perform status check with regular expression on file"""
     cmd = BuildTest(
-        buildspecs=[os.path.join(here, "regex_on_filename.yml"),
-                    os.path.join(here, "regex_on_invalids.yml")],
+        buildspecs=[
+            os.path.join(here, "regex_on_filename.yml"),
+            os.path.join(here, "regex_on_invalids.yml"),
```
Comment (szuananwar) [inline tests/builders/test_builders.py:168]: Please delete the comma , before ]
